Code/Tools/LuaIDE/Source/LUA/LUAEditorContext.cpp

```diff
@@ -1433,7 +1433,9 @@ namespace LUAEditor
             return;
         }
 
+#if !(defined(AZ_PLATFORM_LINUX))
         AZStd::to_lower(const_cast<AZStd::string&>(assetId).begin(), const_cast<AZStd::string&>(assetId).end());
+#endif
```

Comment:
Its not clear why assetId would not ever be all lowercase. The insertion point to m_documentInfoMap in line 1503 will ultimately insert an all lowercase version of asset id. Is there somewhere else the insert is occuring?

Reply:
Ah, I see where the error could be coming from

```
        info.m_assetName = assetId;
        AzFramework::StringFunc::Path::GetFullFileName(assetId.c_str(), info.m_displayName);
        info.m_assetId = assetId;
```
I think maybe we preserve assetId and instead make a lowercase copy of it (assetIdLower?) to use for the m_documentInfoMap lookup and insertion, so that we can keep the original to get the full filename instead

Reply:
isn't it kind of strange to use a lowercase version of the string for ID. things get a bit weird if you try this for lowercase UTF-8 strings. would it make more sense to use the full path un-modified?

Reply:
It looks like the original intent was to make the map into a case-insensitive search. But yeah, its not the ideal way to handle this. I do not know the side effect of not doing a lower-to when the ```info.m_assetName``` and ```info.m_assetId``` is set to assetId, that was why I am suggesting making another non-lowered version just to use for the ```GetFullFileName``` call

Reply:
> It looks like the original intent was to make the map into a case-insensitive search. But yeah, its not the ideal way to handle this. I do not know the side effect of not doing a lower-to when the `info.m_assetName` and `info.m_assetId` is set to assetId, that was why I am suggesting making another non-lowered version just to use for the `GetFullFileName` call

something to investigate further